ext/curl/curl_file.c

```diff
@@ -137,12 +192,17 @@ ZEND_METHOD(CURLFile, setPostFilename)
    Unserialization handler */
 ZEND_METHOD(CURLFile, __wakeup)
 {
-	zend_update_property_string(curl_CURLFile_class, getThis(), "name", sizeof("name")-1, "");
-	zend_throw_exception(NULL, "Unserialization of CURLFile instances is not allowed", 0);
+	zval *fname, rv;
+
+	fname = zend_read_property(curl_CURLFile_class, getThis(), "name", strlen("name"), 1, &rv);
+	if (zval_is_true(fname)) {
+		zend_update_property_string(curl_CURLFile_class, getThis(), "name", sizeof("name")-1, "");
+		zend_throw_exception(NULL, "Unserialization of CURLFile instances with file name is not allowed", 0);
+	}
 }
 /* }}} */
 
-ZEND_BEGIN_ARG_INFO_EX(arginfo_curlfile_create, 0, 0, 1)
+ZEND_BEGIN_ARG_INFO_EX(arginfo_curlfile_ctor, 0, 0, 0)
```

Comment:
curl_file_create in interface.c use same name 'arginfo_curlfile_create' for arg info. So because I added new arg info for function curl_buffer_file_create I decided to rename arg info for constructor.